test/export/StructureDefinition.ExtensionExporter.test.ts

```diff
@@ -216,4 +221,415 @@ describe('ExtensionExporter', () => {
+          isQuoted: false
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toEqual([
+        {
+          type: 'element',
+          expression:
+            'http://example.org/StructureDefinition/mvc-extension#Extension.extension:bar.value[x].extension:secretBar'
+        }
+      ]);
+    });
+
+    it('should set extension context when an alias is used for a resource URL', () => {
+      doc.aliases.set('$PROF', 'http://hl7.org/fhir/StructureDefinition/patient-proficiency');
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: '$PROF#extension[level]',
+          isQuoted: false
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toEqual([
+        {
+          type: 'extension',
+          expression: 'http://hl7.org/fhir/StructureDefinition/patient-proficiency#level'
+        }
+      ]);
+    });
+
+    it('should log an error when no extension or resource can be found with the provided value', () => {
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: 'MysteryExtension',
+          isQuoted: false,
+          sourceInfo: {
+            file: 'Context.fsh',
+            location: {
+              startLine: 8,
+              startColumn: 7,
+              endLine: 8,
+              endColumn: 25
+            }
+          }
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toBeUndefined();
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        'Could not find resource MysteryExtension to use as extension context.'
+      );
+      expect(loggerSpy.getLastMessage('error')).toMatch(/File: Context\.fsh.*Line: 8\D*/s);
+    });
+
+    describe('#withCustomResource', () => {
+      beforeEach(() => {
+        const myObservation = new Profile('MyObservation');
+        myObservation.parent = 'Observation';
+        myObservation.id = 'my-obs';
+        doc.profiles.set(myObservation.name, myObservation);
+      });
+
+      it('should set extension context for a custom resource root element by id', () => {
+        const extension = new Extension('MyExtension');
+        extension.contexts = [
+          {
+            value: 'my-obs',
+            isQuoted: false
+          }
+        ];
+        doc.extensions.set(extension.name, extension);
+        const exported = exporter.exportStructDef(extension);
+        expect(exported.context).toEqual([
+          {
+            type: 'element',
+            expression: 'http://hl7.org/fhir/us/minimal/StructureDefinition/my-obs#Observation'
```

Comment:
Oh... interesting. I was going to ask why, but now I see. The spec says:
> If there is no #, the Element id is one defined in the base specification.

Good call.